src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/ConnectionManager.java

```diff
@@ -165,45 +185,29 @@ public void handleTopologyRecoveryException(Connection conn, Channel ch, Topolog
             }
 
-            private void turnOffReadiness(Throwable exception){
+            private void turnOffReadiness(Throwable exception) {
                 metrics.getReadinessMonitor().disable();
                 LOGGER.debug("Set RabbitMQ readiness to false. RabbitMQ error", exception);
             }
         });
 
-        factory.setAutomaticRecoveryEnabled(true);
-        factory.setConnectionRecoveryTriggeringCondition(shutdownSignal -> {
-            if (connectionIsClosed.get()) {
-                return false;
-            }
-
-            int tmpCountTriesToRecovery = connectionRecoveryAttempts.get();
-
-            if (tmpCountTriesToRecovery < connectionManagerConfiguration.getMaxRecoveryAttempts()) {
-                LOGGER.info("Try to recovery connection to RabbitMQ. Count tries = {}", tmpCountTriesToRecovery + 1);
-                return true;
-            }
-            LOGGER.error("Can not connect to RabbitMQ. Count tries = {}", tmpCountTriesToRecovery);
-            if (onFailedRecoveryConnection != null) {
-                onFailedRecoveryConnection.run();
-            } else {
-                // TODO: we should stop the execution of the application. Don't use System.exit!!!
-                throw new IllegalStateException("Cannot recover connection to RabbitMQ");
-            }
-            return false;
-        });
+        factory.setAutomaticRecoveryEnabled(configuration.isAutomaticRecoveryEnabled());
+        factory.setConnectionRecoveryTriggeringCondition(shutdownSignal -> !connectionIsClosed.get());
 
         factory.setRecoveryDelayHandler(recoveryAttempts -> {
-                    int tmpCountTriesToRecovery = connectionRecoveryAttempts.getAndIncrement();
-
-                    int recoveryDelay = connectionManagerConfiguration.getMinConnectionRecoveryTimeout()
-                            + (connectionManagerConfiguration.getMaxRecoveryAttempts() > 1
-                                ? (connectionManagerConfiguration.getMaxConnectionRecoveryTimeout() - connectionManagerConfiguration.getMinConnectionRecoveryTimeout())
-                                    / (connectionManagerConfiguration.getMaxRecoveryAttempts() - 1)
-                                    * tmpCountTriesToRecovery
-                                : 0);
+                    int minTime = connectionManagerConfiguration.getMinConnectionRecoveryTimeout();
+                    int maxTime = connectionManagerConfiguration.getMaxConnectionRecoveryTimeout();
+                    int maxRecoveryAttempts = connectionManagerConfiguration.getMaxRecoveryAttempts();
+                    int deviationPercent = connectionManagerConfiguration.getRetryTimeDeviationPercent();
+
+                    LOGGER.debug("Try to recovery connection to RabbitMQ. Count tries = {}", recoveryAttempts);
+                    int recoveryDelay = RetryingDelay.Companion.getRecoveryDelay(recoveryAttempts, minTime, maxTime, maxRecoveryAttempts, deviationPercent);
```

Comment:
If you added `@JvmStatic` to the method you now can remove access to the `Companion` object from the method call